srcs/vm/ops/add.c

```diff
@@ -0,0 +1,46 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        ::::::::            */
+/*   add.c                                              :+:    :+:            */
+/*                                                     +:+                    */
+/*   By: mvan-eng <mvan-eng@student.codam.nl>         +#+                     */
+/*                                                   +#+                      */
+/*   Created: 2020/08/09 19:53:23 by mvan-eng      #+#    #+#                 */
+/*   Updated: 2020/08/13 15:29:39 by merlijn       ########   odam.nl         */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "vm.h"
+
+static int	check_add_argument_types(t_argument *args)
+{
+	if (args[0].type != ARG_TYPE_REG
+	|| args[1].type != ARG_TYPE_REG
+	|| args[2].type != ARG_TYPE_REG)
+		return (ERROR);
+	return (OK);
+}
+
+void		add(char *mem, t_cursor *cursor)
+{
+	int		arg1;
+	int		arg2;
+	int		arg3;
+	int		sum;
+
+	get_argument_types(mem, cursor);
+	if (check_add_argument_types(cursor->args) == ERROR)
+		return ;
+	arg1 = mem[(cursor->pos + 2) % MEM_SIZE];
```

Comment:
Maybe nice to write a function that does the modules thingies:

```get_arena_pos(int pos)```

Reply:
Will do

Reply:
``` C
void *get_arena_byte(void *mem, int pos)
arg1 = *get_arena_byte(mem, cursor->pos + 2)
```

Reply:
`int		get_pos(int cursor_pos, int pos)
{
	return ((cursor_pos + pos) % MEM_SIZE);
}
`
Zou zo handiger zijn, get_direct_argument moet 2 of 4 bytes samenvoegen dus functie die positie berekent in plaats van 1 byte is universeler.